test/pstl_offload/memory/system_allocations.h

```diff
@@ -0,0 +1,148 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _SYSTEM_ALLOCATIONS_H
+#define _SYSTEM_ALLOCATIONS_H
+
+extern "C" {
+struct _IO_FILE;
+typedef struct _IO_FILE FILE;
+
+extern FILE *stderr;
+int fprintf(FILE *stream, const char *format, ...);
+void exit(int status);
+}
+
+// to have native allocation, we must not include possibly overloaded header
+#if DO_STD_INJECTION
+namespace std {
```

Comment:
Could you please explain, what is the reason for not including any C++ standard headers into this file?
If we want a non-replaced `malloc` functions in `perform_system_allocations`, we can either build the second TU without `fsycl-pstl-offload` option or define the `TOP_LEVEL` guarding macro before including `cstdlib` or `new`.

I guess in this case, these strange forward declarations can be removed

Reply:
Thanks! Guarding macro is much better.